### ISSUE TITLE
Assertion failure when copying lots of lines

### ISSUE THREAD

**Issue by hqkhan** [ibhagwan/smartyank.nvim#5]
Hi IBhagwan. I'm seeing a lot of assertion failures if I try to yank lots of lines (250+).

```
2681 lines yanked
Error detected while processing TextYankPost Autocommands for "*":
Error executing lua callback: ...te/pack/packer/opt/smartyank.nvim/lua/smartyank/init.lua:49: assertion failed!
stack traceback:
        [C]: in function 'assert'
        ...te/pack/packer/opt/smartyank.nvim/lua/smartyank/init.lua:49: in function 'osc52printf'
        ...te/pack/packer/opt/smartyank.nvim/lua/smartyank/init.lua:82: in function 'yank'
        ...te/pack/packer/opt/smartyank.nvim/lua/smartyank/init.lua:147: in function <...te/pack/packer/opt/smartyank.nvim/lua/smartyank/init.lua:138>
```

This is pointing to this [line](https://github.com/ibhagwan/smartyank.nvim/blob/master/lua/smartyank/init.lua#L49). Even though there is an assertion failure, the lines still get copied. 

I'm on `NVIM v0.8.2-dev-29+gda68023d9`.

**Comment by ibhagwan** [comment]
I can certainly remove the assert, it’s just for debugging, let me first investigate why is `chansend` not returning the right number of bytes copied.

Is this a new issue with `8.2`? Does this happen only for large yank operations?

**Comment by ibhagwan** [comment]
Tested, seems to work correctly even when `chansend` returns 0 bytes copied, not sure why, assert was removed so it shouldn't fail anymore.

**Comment by hqkhan** [comment]
Ty, it's good now. Yeah, potentially a 0.7 vs 0.8 bug? I was using 0.7 before. I did only had it happen on large yank operations and never on a smaller one. It was a bit flaky tbh. I tried yanking say 250 lines and had assertion error. Tried again and it'd randomly go through. Try again and it runs into the assertion error. More often than not, it was getting an error. It was rare for it to pass the assertion.
